glean-core/src/metrics/mod.rs

```diff
@@ -165,7 +169,40 @@ pub trait MetricType {
     /// This depends on the metrics own state, as determined by its metadata,
     /// and whether upload is enabled on the Glean object.
     fn should_record(&self, glean: &Glean) -> bool {
-        glean.is_upload_enabled() && self.meta().should_record()
+        if !glean.is_upload_enabled() {
+            return false;
+        }
+
+        // Get the current disabled field from the metric metadata, including
+        // the encoded Nimbus epoch
+        let disabled_field = self.meta().disabled.load(Ordering::Relaxed);
+        // Grab the epoch from the upper nibble
+        let epoch = disabled_field >> 4;
+        // Get the disabled flag from the lower nibble
+        let disabled = disabled_field & 0xF;
+        // Get the current Nimbus epoch to see if we need to bother with the
+        // more expensive HashMap lookup
+        let nimbus_epoch = glean.nimbus_epoch.load(Ordering::Acquire);
+        if epoch == nimbus_epoch {
+            return disabled == 0;
+        }
+        // The epoch's didn't match so we need to look up the disabled flag
+        // by the base_identifier from the in-memory HashMap
+        let metrics_disabled = &glean.nimbus_metrics_config.lock().unwrap().metrics_disabled;
+        // Get the value from the remote configuration if it is there, otherwise return the default value.
+        let current_disabled =
+            if let Some(is_disabled) = metrics_disabled.get(&self.meta().base_identifier()) {
+                u8::from(*is_disabled)
+            } else {
+                u8::from(self.meta().defaults.disabled)
+            };
+
+        // Re-encode the epoch and enabled status and update the metadata
+        let new_disabled = (nimbus_epoch << 4) | (current_disabled & 0xF);
+        self.meta().disabled.store(new_disabled, Ordering::Relaxed);
+
+        // Return the bool form of whether or not the metric is disabled
```

Comment:
Heh, reading this I got confused again. The method is `should_record`, so exactly the opposite of `is disabled?`.

```suggestion
        // Return the bool form of whether or not the metric should be recorded
```